### PR TITLE
Structural breaks

### PR DESCRIPTION
**Pull request by proskurin**
# Description
Add SADF (linear + sub-martingale), CUSUM structural breaks tests

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- test_structural_breaks.py

**Test Configuration**:
* PyCharm
* Ubuntu 16.04

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/structural_breaks/chow.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    Multithread implementation of Chow-Type Dickey-Fuller Test, p.251-252
+
+    :param series: (pd.Series) series to test
+    :param min_length: (int) minimum sample length used to estimate statistics
+    :param num_threads: (int): number of cores to use
+    :return: (pd.Series) of Chow-Type Dickey-Fuller Test statistics
+    """
+    # Indices to test. We drop min_length first and last values
+    molecule = series.index[min_length:series.shape[0] - min_length]
```
Comment (artem-matiash) [inline mlfinlab/structural_breaks/chow.py]: If I understand correctly, input series index is pd.DateTimeIndex, hence, int slice won't work.

File: mlfinlab/structural_breaks/cusum.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            elif test_type == 'two_sided':
+                values_diff = abs(series.loc[index] - series.loc[ind])
+            else:
+                raise ValueError('Test type is unknown: can be either one_sided or two_sided')
+
+            temp_integer_index = series_t.index.get_loc(ind)
+            s_n_t = 1 / (sigma_sq_t * np.sqrt(integer_index - temp_integer_index)) * values_diff
+            if s_n_t > max_s_n_value:
+                max_s_n_value = s_n_t
+                max_s_n_critical_value = np.sqrt(4.6 + np.log(integer_index - temp_integer_index))
```
Comment (artem-matiash) [inline mlfinlab/structural_breaks/cusum.py]: Please, document what 4.6 is and it would be perfect to define it as CONST.

Comment (Jackal08) [reply mlfinlab/structural_breaks/cusum.py]: What is 4.6?

Comment (proskurin) [reply mlfinlab/structural_breaks/cusum.py]: This is the empirical constant from the book. Let me comment on that

**Pull request merged**